articles/create-package.md

```diff
@@ -5,18 +5,92 @@ title: "Creating a Package"
-6. It is also possible to debug our code while it is running by setting breakpoints, either by clicking on the left of the line we want to debug or by hitting **F9** over the target code. After this, you can run the code step by step and inspect the runtime value of variables. ![Setting a breakpoint on the sine transform](~/images/extensions-breakpoint.png) ![Debugging the sine transform](~/images/extensions-debugging.png)
+
+2. Select the **Create a new project** option. In the **Create a new project** dialog, type `bonsai` in the search toolbox. The project template for creating a Bonsai package should now be displayed.
+
+    ![Creating a new Bonsai package project](~/images/extensions-packageproject.png)
+
+3. Give the project a name and a location, and press the `Create` button. After the project is created, you should see that a file "**Source1.cs**" has been added to the solution explorer. This file contains an example implementation of a custom source.
+
+    ![Bonsai source template](~/images/extensions-sourcetemplate.png)
+
+4. Every Bonsai operator specifies an observable sequence using the <xref href="System.IObservable`1"/> interface. The [System.Reactive](http://reactivex.io/) package provides a comprehensive library of methods used to generate and manipulate observable sequences in C#. The simplest way to implement a source is by using the methods in the <xref href="System.Reactive.Linq.Observable"/> class.
+
+    As an example, the expression below will generate a simple periodic sinewave by applying the projection operator <xref href="System.Reactive.Linq.Observable.Select*"/> to the incremental counter sequence generated by a <xref href="System.Reactive.Linq.Observable.Timer*"/>.
+
+    ```c#
+    [Description("")]
+    [Combinator(MethodName = nameof(Generate))]
+    [WorkflowElementCategory(ElementCategory.Source)]
+    public class Source1
+    {
+        public IObservable<double> Generate()
+        {
+            return Observable
+                .Timer(
+                    dueTime: TimeSpan.Zero,
+                    period: TimeSpan.FromSeconds(0.5))
+                .Select(counter => Math.Sin(counter));
+        }
+    }
+    ```
+
+5. We can test our operator by starting the project with **F5**. This will automatically launch the Bonsai editor with our package preloaded in the **Toolbox**. Add the new `Source1` node and run the workflow.
+
+    ![Running the sine source](~/images/extensions-running.png)
+
+6. If we need to parameterize our operator, we can add new public properties to the class. Custom editors can be used to provide interactive widgets for editing the property values. Several common widgets are provided in the <xref href="Bonsai.DesignTypes"/> class. For example, to make the period customizable with a slider over a specified range, we can modify our source operator:
```

Comment:
Not sure if the old "full prop" syntax is not better for the properties... This syntax doesn't seem to work for extensions cause they are still using a older version of .NET.
I understand that it still works for packages but still it might be better to write a syntax that works for both?

Reply:
Yes, I was a bit torn on this, but I agree at least for examples' sake it might be nice to have a common workable syntax. It's easy enough to figure out the newer syntax by "hands-on" exposure.

Reply:
There is another issue however, which is more "annoying". The default source placed by the template uses the `nameof` operator to point the combinator to the `Generate` method. Unfortunately `nameof` syntax is also not compatible with the old scripting.

Might be better to just leave the script as-is at this point and maybe do the complementary step of explaining the use of legacy C# in scripting extensions. Whoever starts using it more seriously will quickly hit this wall anyway, and eventually we will migrate to Roslyn and have full C# compatibility.